Physicist/Physicist/Actor/Actor.cs

```diff
@@ -297,5 +342,79 @@ protected virtual bool OnCollision(Fixture fixtureA, Fixture fixtureB, FarseerPh
 
             return this.IsEnabled;
         }
+
+        protected void PlaySound(string name)
+        {
+            List<string> removelist = new List<string>();
+
+            // check for every Guid in the sound controller.
+            for (int i = 0; i < this.PlayingSounds.Count; i++)
```

Comment:
for (int i = 0 this.PlayingSounds.Count - 1; i >= 0; i--)
{
    var sound = this.PlayingSounds.ElementAt(i).Value;
    if(!SoundController.ContainsInstance(sound))
    {
         this.PlayingSounds.remove(sound);
    }
}